### PR TITLE
Multithreaded judgement support

### PR DESCRIPTION
**Pull request by tahaeghtesad**
Added multithreading judge on single server with multiple cors.

### CODE REVIEW

**Comment by tahaeghtesad** [comment]
sorry, but I accidentally deleted "get_checker()" method..

File: application/controllers/grader.php [hunk 1]
```diff
@@ -43,15 +50,13 @@ public function run()
...
-
-			$this->check_in();
-			if ( ! $this->has_new_submission())
-				continue;
-
-			$submission = $this->get_new_submission();
-			$this->grade_submission($submission);
-			
-			$this->logger->log('grader', 'Waiting for new submission...');
+			$this->check_in($number);
```
Comment (sokokaleb) [inline application/controllers/grader.php]: The function `check_in` below doesn't have the parameter `number`, but this function call passes it.
Are you sure this is working?

**Comment by fushar** [comment]
@tahaeghtesad: We just rewrote the whole git history for Regrader 2. Please rebase/rework your changes accordingly. Thanks!

**Comment by fushar** [comment]
Dear @tahaeghtesad, could you resolve my and @sokokaleb's issues please? If you don't have time, please inform me so that I will resolve myself. Thanks!

**Comment by tahaeghtesad** [comment]
Hi @fushar ;)
I was completely free this summer, and wondering how could I fill it with useful activities :D
But unfortunately I have forgotten about Regrader and the mess that I have requested to pull. The problematic pull request I was making, was caused by making changes on another computer and not having all the files completely. ( This multithreaded judgment has more files that should be pulled other than grader.php, and I have forgotten to add others )
I will be very happy to help you on this, more than this just this issue.
Now, I have more than a month completely free to work on this, and other parts as well ( I have translated the site to Persian and, I think it would be useful to add it to Regrader 2. )

You can contact me by email at : tahaeghtesad@gmail.com or Telegram by +989353361064.
We can discuss issues and more there.

Thanks for concerns.
Taha Eghtesad

**Comment by fushar** [comment]
Nice to hear that! That's much more than I expected! 

Then, first thing first, let's finish this feature, shall we? :)

Since your fork has diverged from the main repository, here is a quick list of things you would have to do for this pull request:

- Re-clone fushar/regrader.
- Try running it (see the README).
- Re-do your work, including the missing files.
- Push-force to tahaeghtesad/master.

Thanks for your time! This feature should be really useful for the others :)

**Pull request closed without merging**